From 69096bb1afed435e2589c525ed30a17ac56865a7 Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@freescale.com>
Date: Fri, 21 Dec 2012 13:46:10 -0600
Subject: [PATCH 134/304] ENGR00238059: iMX53: Align iMX53 SMD Jellybean
 device with other JB devices.

Align iMX53 SMD Jellybean device config with other Google devices.
Also add wifi support URLs.

Signed-off-by: Nitin Garg <nitin.garg@freescale.com>
---
 .../frameworks/base/core/res/res/values/config.xml | 25 +++++++++++++++++++++-
 .../packages/apps/Launcher2/res/values/config.xml  |  3 +++
 .../packages/apps/Settings/res/values/strings.xml  | 22 +++++++++++++++++++
 imx5x/imx53_smd.mk                                 |  8 +++----
 imx5x/imx5x.mk                                     | 20 +++++++++++++++--
 imx5x/init.freescale.usb.rc                        |  7 ++++++
 imx5x/init.rc                                      | 12 +++++++++++
 7 files changed, 90 insertions(+), 7 deletions(-)
 create mode 100644 imx53_smd/overlay/packages/apps/Launcher2/res/values/config.xml
 create mode 100644 imx53_smd/overlay/packages/apps/Settings/res/values/strings.xml

diff --git a/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml b/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml
index eea17b0..cc2fbe4 100755
--- a/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/imx53_smd/overlay/frameworks/base/core/res/res/values/config.xml
@@ -20,6 +20,8 @@
 <!-- These resources are around just to allow their values to be customized
      for different hardware and product builds. -->
 <resources>
+    <!-- Software blur is too slow on this device. -->
+    <bool name="config_sf_slowBlur">true</bool>
 
     <!-- List of regexpressions describing the interface (if any) that represent tetherable
          USB interfaces.  If the device doesn't want to support tething over USB this should
@@ -168,8 +170,29 @@
     <!-- Minimum screen brightness allowed by the power manager. -->
     <integer name="config_screenBrightnessDim">25</integer>
 
+    <!-- This device is not "voice capable"; it's data-only. -->
+    <bool name="config_voice_capable">false</bool>
+
+    <!-- Number of database connections opened and managed by framework layer
+         to handle queries on each database. -->
+    <integer name="db_connection_pool_size">3</integer>
+
+    <!-- Max space (in MB) allocated to DownloadManager to store the downloaded files
+         if they are to be stored in DownloadManager's data dir,
+         which typically is /data/data/com.android.providers.downloads/files. -->
+    <integer name="config_downloadDataDirSize">200</integer>
+
+    <!-- When the free space available in DownloadManager's data dir falls
+         below the percentage value specified by this param, DownloadManager
+         starts removing files to try to make percentage of available
+         free space above this threshold value. -->
+    <integer name="config_downloadDataDirLowSpaceThreshold">10</integer>
+
+    <!-- When a database query is executed, the results retuned are paginated
+         in pages of size (in KB) indicated by this value -->
+    <integer name="config_cursorWindowSize">2048</integer>
+
     <!-- Base "touch slop" value used by ViewConfiguration as a
          movement threshold where scrolling should begin. -->
     <dimen name="config_viewConfigurationTouchSlop">24dp</dimen>
-
 </resources>
diff --git a/imx53_smd/overlay/packages/apps/Launcher2/res/values/config.xml b/imx53_smd/overlay/packages/apps/Launcher2/res/values/config.xml
new file mode 100644
index 0000000..3b82809
--- /dev/null
+++ b/imx53_smd/overlay/packages/apps/Launcher2/res/values/config.xml
@@ -0,0 +1,3 @@
+<resources>
+    <bool name="config_largeHeap">true</bool>
+</resources>
diff --git a/imx53_smd/overlay/packages/apps/Settings/res/values/strings.xml b/imx53_smd/overlay/packages/apps/Settings/res/values/strings.xml
new file mode 100644
index 0000000..00f0cda
--- /dev/null
+++ b/imx53_smd/overlay/packages/apps/Settings/res/values/strings.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2007 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!--  Help URLs for some screens. Not specified here. Specified in product overlays --><skip/>
+    <!-- Help menu label [CHAR LIMIT=20] -->
+    <!-- Help URL, WiFi [DO NOT TRANSLATE] -->
+    <string name="help_url_wifi" translatable="false">http://support.google.com/wifi</string>
+    <string name="help_url_accounts" translatable="false">http://support.google.com/accounts</string>
+</resources>
diff --git a/imx5x/imx53_smd.mk b/imx5x/imx53_smd.mk
index 3855ae2..8df04ab 100644
--- a/imx5x/imx53_smd.mk
+++ b/imx5x/imx53_smd.mk
@@ -23,12 +23,10 @@ PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
     frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
     frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
-    frameworks/native/data/etc/android.hardware.sensor.accelerometer.xml:system/etc/permissions/android.hardware.sensor.accelerometer.xml \
-    frameworks/native/data/etc/android.hardware.sensor.compass.xml:system/etc/permissions/android.hardware.sensor.compass.xml \
+    frameworks/native/data/etc/android.hardware.sensor.barometer.xml:system/etc/permissions/android.hardware.sensor.barometer.xml \
     frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
     frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
     frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml \
-    frameworks/native/data/etc/android.hardware.telephony.gsm.xml:system/etc/permissions/android.hardware.telephony.gsm.xml \
     frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml
 
 # for PDK build, include only when the dir exists
@@ -43,7 +41,9 @@ PRODUCT_COPY_FILES += \
 	device/fsl/imx53_smd/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/imx53_smd/mixer_paths.xml:system/etc/mixer_paths.xml
 
-PRODUCT_PREF_AAPT_CONFIG := mdpi
+# Bluetooth config file
+PRODUCT_COPY_FILES += \
+	system/bluetooth/data/main.nonsmartphone.conf:system/etc/bluetooth/main.conf
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/imx53_smd/overlay
 
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 7716ffa..0afc73a 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -1,6 +1,7 @@
 $(call inherit-product, $(SRC_TARGET_DIR)/product/languages_full.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/generic.mk)
-$(call inherit-product, frameworks/base/data/sounds/OriginalAudio.mk)
+$(call inherit-product, frameworks/base/data/sounds/AudioPackage4.mk)
+$(call inherit-product, frameworks/base/data/sounds/AudioPackage5.mk)
 $(call inherit-product, frameworks/base/data/fonts/fonts.mk)
 $(call inherit-product, frameworks/base/data/keyboards/keyboards.mk)
 
@@ -75,6 +76,7 @@ PRODUCT_PACKAGES += \
 	copybit.imx5x				\
 	hwcomposer.imx5x            		\
 	camera.imx5x            		\
+	power.imx5x				\
 	magd
 
 # Bluetooth firmware files.
@@ -251,6 +253,14 @@ PRODUCT_PACKAGES += $(omx_libs) $(omx_excluded_libs)
 
 PRODUCT_PACKAGES += libubi ubinize ubiformat ubiattach ubidetach ubiupdatevol ubimkvol ubinfo mkfs.ubifs
 
+# FUSE based emulated sdcard daemon
+PRODUCT_PACKAGES += sdcard
+
+# ntfs-3g binary
+PRODUCT_PACKAGES += \
+	ntfs-3g		\
+	ntfsfix
+
 # for CtsVerifier
 PRODUCT_PACKAGES += \
     com.android.future.usb.accessory
@@ -267,6 +277,12 @@ PRODUCT_COPY_FILES +=	\
 	device/fsl/imx5x/init.gprs-pppd:system/etc/ppp/init.gprs-pppd \
 	device/fsl/imx5x/initlogo.rle:root/initlogo.rle
 
+# for bugmailer
+PRODUCT_PACKAGES += send_bug
+PRODUCT_COPY_FILES += \
+    system/extras/bugmailer/bugmailer.sh:system/bin/bugmailer.sh \
+    system/extras/bugmailer/send_bug:system/bin/send_bug
+
 # VPU firmware
 PRODUCT_COPY_FILES +=	\
 	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx53.bin:system/lib/firmware/vpu/vpu_fw_imx53.bin
@@ -281,5 +297,5 @@ PRODUCT_DEFAULT_DEV_CERTIFICATE := \
         device/fsl/common/security/testkey
 
 # include a google recommand heap config file.
-include frameworks/native/build/tablet-dalvik-heap.mk
+include frameworks/native/build/tablet-7in-hdpi-1024-dalvik-heap.mk
 
diff --git a/imx5x/init.freescale.usb.rc b/imx5x/init.freescale.usb.rc
index 1e331de..1aa2907 100644
--- a/imx5x/init.freescale.usb.rc
+++ b/imx5x/init.freescale.usb.rc
@@ -1,3 +1,10 @@
+on init
+    write /sys/class/android_usb/android0/iSerial ${ro.serialno}
+
+on boot
+    write /sys/class/android_usb/android0/iManufacturer ${ro.product.manufacturer}
+    write /sys/class/android_usb/android0/iProduct ${ro.product.model}
+
 # USB massstorage configuration, with adb
 on property:sys.usb.config=mass_storage,adb
     write /sys/class/android_usb/android0/enable 0
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 81cf243..ad79ba7 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -363,6 +363,10 @@ on boot
 # Set this property so surfaceflinger is not started by system_init
     setprop system_init.startsurfaceflinger 0
 
+# Default Read Ahead value for sdcards
+    write /sys/block/mmcblk0/queue/read_ahead_kb 2048
+    write /sys/block/mmcblk1/queue/read_ahead_kb 2048
+
     class_start core
     class_start main
 
@@ -575,3 +579,11 @@ service mdnsd /system/bin/mdnsd
     socket mdnsd stream 0660 mdnsr inet
     disabled
     oneshot
+
+# bugreport is triggered by the HOME and VOLUME-UP keys
+service bugreport /system/bin/bugmailer.sh -b /system/media/audio/ui/Dock.ogg -e /system/media/audio/ui/Undock.ogg
+    class main
+    disabled
+    oneshot
+    keycodes 115 102
+
-- 
1.8.0

